Sources/SwiftLexicalLookup/Scopes/ScopeImplementations.swift

```diff
@@ -665,13 +671,16 @@ import SwiftSyntax
       )
     }
 
+    let lookInMembers: [LookupResult] =
+      inheritanceClause?.range.contains(lookUpPosition) ?? false ? [] : [.lookInMembers(self)]
```

Comment:
Instead of blacklisting the inheritance clause, wouldn’t it be better to whitelist the `memberBlock`? I got to this because I was thinking that `self` shouldn’t be available in the generic parameters of the type.

Reply:
That's a good point. As for now, I couldn't find a specific case this wouldn't work and it definitely improves readability of this part of code as well. I changed it to check if lookup was started inside `memberBlock`.